test/gen_integration.sh

```diff
@@ -329,7 +373,31 @@ languageVariables:
           action: deploy
           manifests: \${{ steps.bake.outputs.manifestsBundle }}
           images: |
-            testapp:curr
+            $imagename
+      - name: Wait for rollout
+        continue-on-error: true
+        id: rollout
+        run: |
+          kubectl rollout status deployment/testapp --timeout=2m
+      - name: Print K8s Objects
+        run: |
+          kubectl get po -o json
+          kubectl get svc -o json
+          kubectl get deploy -o json
+      - name: Curl Endpoint
+        run: |
+          kubectl get svc
+          echo 'Starting minikube tunnel'
+          minikube tunnel  > /dev/null 2>&1 & tunnelPID=\$!
+          sleep 120
+          kubectl get svc
+          SERVICEIP=\$(kubectl get svc -o jsonpath={'.items[1].status.loadBalancer.ingress[0].ip'})
+          echo \"SERVICEIP: \$SERVICEIP\"
```

Comment:
yep this script generates the github action script, so it needs to escape so that it is parsed at github action run time. We should consider refactoring this gen script in another PR you're right that this is confusing